schemas/string_interpolation/transforms/generate_date_range.json

```diff
@@ -0,0 +1,74 @@
+{
+  "$id": "https://eq.ons.gov.uk/string_interpolation/transforms/generate_date_range.json",
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "generate_date_range": {
+    "description": "A transform for generating the start and end date of a date range from a reference date, offset number of weeks, number of days in range and an optional first day of the week.",
+    "type": "object",
+    "properties": {
+      "transform": {
+        "type": "string",
+        "enum": ["generate_date_range", "format_date_range_pair"]
+      },
+      "arguments": {
+        "type": "object",
+        "properties": {
+          "reference_date": {
+            "oneOf": [
+              {
+                "$ref": "https://eq.ons.gov.uk/string_interpolation/definitions.json#/value_sources"
+              },
```

Comment:
For the reference day, we need to validate that when an `answer` source is used, it is one of our date input answer types, i.e reference a text field answer should be invalid but would currently pass validation.

Reply:
From our convo this is out of scope for this one? Did you intend to raise/raise a card?

Reply:
Yup, a card has been raised to address that.